docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -2278,7 +2278,7 @@ Otherwise, Quarkus attempts to monitor the extension classes and this may result
 
 [source,bash]
 ----
-./mvnw compile quarkus:dev -DnoDeps
```

Comment:
This paragraph is dedicated to multimodule projects and this is where we should mention when explicit `compile` is recommended. This paragraph is also outdated. Our bootstrap has evolved since those days and is a lot more reliable today. `-DnoDeps` isn't really *required* today.

Reply:
+1 on mentioning `compile`, maybe even `test-compile` for continuous testing.

PS: Because of a error when using ECJ, I had to resort to specifying an explicit goal, so that devmojo doesn't run those executions on its own (because this is where the error happens).

Reply:
@aloubyansky I think it would probably be easier if you added an additional commit on top of @maxandersen 's changes.

Reply:
@famod what's that error you get with each? Got issue link?

@gsmet sorry. GitHub notifications sucks for me. Will touch this up while travelling today.

Reply:
@aloubyansky I've reworked this section to talk about need for explicit compile and removed noDeps mention. Should it be mentioned somewhere else or is noDeps just not relevant to use anymore ?

Reply:
By default we make as many modules in the workspace reloadable as we can handle (exceptions are extension artifacts, dependencies of extensions, parent-first artifacts and dependencies of parent-first).
`noDeps` makes only the app root module reloadable.